### ISSUE TITLE
feat(prefer-find-by): report `waitFor` with presence matchers

### ISSUE THREAD

**Issue by zaicevas** [testing-library/eslint-plugin-testing-library#420]
Originally mentioned in https://github.com/testing-library/eslint-plugin-testing-library/issues/411#issuecomment-882109271

`prefer-find-by` could report presence matchers (`.toBeInTheDocument()`, `toBeDefined()`, etc) that are used in conjunction with RTL queries. Currently, `prefer-find-by` does not report such cases:

```
    await waitFor(() => screen.getByText('button')) // reported
    await waitFor(() => expect(screen.getByText('button')).toBeInTheDocument()) // not reported
```
